### PR TITLE
Get time metadata from ledger state

### PR DESCRIPTION
**Pull request by kderme**
https://jira.iohk.io/browse/CAD-2796

### CODE REVIEW

File: cardano-db-sync-extended/src/Cardano/DbSync/Plugin/Extended.hs [hunk 1]
```diff
@@ -19,7 +19,9 @@ extendedDbSyncNodePlugin backend =
               ++ [epochPluginOnStartup backend]
         , plugInsertBlock =
             plugInsertBlock defPlugin
-                ++ [epochPluginInsertBlock backend]
```
Comment (erikd) [inline cardano-db-sync-extended/src/Cardano/DbSync/Plugin/Extended.hs:22]: Does not seem correct for that to be removed.

File: cardano-sync/src/Cardano/Sync.hs [hunk 2]
```diff
@@ -276,9 +268,9 @@ dbSyncProtocols trce env metricsSetters plugin queryVar runDBThreadFunction vers
     localStateQuery :: RunMiniProtocol 'InitiatorMode BSL.ByteString IO () Void
     localStateQuery =
       InitiatorProtocolOnly $ MuxPeer
-        (contramap (Text.pack . show) . toLogObject $ appendName "local-state-query" trce)
+        Logging.nullTracer
         (cStateQueryCodec codecs)
-        (localStateQueryClientPeer (localStateQueryHandler queryVar))
+        localStateQueryPeerNull
```
Comment (erikd) [inline cardano-sync/src/Cardano/Sync.hs:273]: Why? I suspect there is useful logging coming out of those two loggers.

File: cardano-sync/src/Cardano/Sync/Plugin.hs [hunk 3]
```diff
@@ -36,8 +38,8 @@ data SyncNodePlugin = SyncNodePlugin
     -- This will not be called for the original genesis block, but will be called for
     -- all subsequent blocks.
     -- Blocks (including epoch boundary blocks) are called in sequence from the oldest to the newest.
-  , plugInsertBlock :: [Trace IO Text -> SyncEnv -> [BlockDetails] -> IO (Either SyncNodeError ())]
-
+  , plugInsertBlock :: Trace IO Text -> SyncEnv -> [CardanoBlock] -> ExceptT SyncNodeError IO [BlockDetails]
```
Comment (erikd) [inline cardano-sync/src/Cardano/Sync/Plugin.hs:41]: Why? This seems to reduce the generality of the "plugin" system.

**Review by erikd — COMMENTED** [review]
A couple of questions on some of the seemingly unrelated changes.

File: cardano-db-sync-extended/src/Cardano/DbSync/Plugin/Extended.hs (hunk 1 above)
Comment (kderme) [reply cardano-db-sync-extended/src/Cardano/DbSync/Plugin/Extended.hs:22]: It is moved to `plugInsertBlockDetails`

File: cardano-sync/src/Cardano/Sync.hs (hunk 2 above)
Comment (kderme) [reply cardano-sync/src/Cardano/Sync.hs:273]: we no longer use the local state query

File: cardano-sync/src/Cardano/Sync/Plugin.hs (hunk 3 above)
Comment (kderme) [reply cardano-sync/src/Cardano/Sync/Plugin.hs:41]: The idea is that `plugInsertBlock` also creates the `[BlockDetails]`. `plugInsertBlockDetails` makes sure that we don't reduce the generality of the "plugin" system.

**Comment by ksaric** [comment]
Judging from what I see, the issue I have is not related to this change, and this plugin change will be backward compatible with `smash` changes.

File: cardano-sync/src/Cardano/Sync/Plugin.hs (hunk 3 above)
Comment (erikd) [reply cardano-sync/src/Cardano/Sync/Plugin.hs:41]: What is the difference between `plugInsertBlock` and `plugInsertBlockDetails` ?

Seems the only difference is `CardanoBlock` vs `BlockDetails`. Can these be used interchangably? Can we do without the older version?

Comment (kderme) [reply cardano-sync/src/Cardano/Sync/Plugin.hs:41]: `plugInsertBlock` applies the block to the ledger state and creates the `BlockDetails`. So `plugInsertBlockDetails` has the details available to do any extra work with them.

Comment (erikd) [reply cardano-sync/src/Cardano/Sync/Plugin.hs:41]: Ah ok, now I get it. I will add some comments to make that more clear.

Comment (erikd) [reply cardano-sync/src/Cardano/Sync/Plugin.hs:41]: Maybe `plugInsertBlock` should even be renamed. Maybe something like `pluginPrepareBlocks` .

**Pull request closed without merging**